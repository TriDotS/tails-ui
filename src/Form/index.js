import React from 'react'
import styled from 'styled-components'

export const Form = styled.fieldset`
  padding: 20px;
`

export default ({ children, label }) =>
  <Form>
    <legend>{label}</legend>
    {children}
  </Form>
