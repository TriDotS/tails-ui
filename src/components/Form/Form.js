import React from 'react'
import styled from 'styled-components'

export const Form = styled.form`padding: 0px;`

export default ({ children, label, bordered, onSubmit }) =>
  <Form onSubmit={onSubmit}>
    {bordered
      ? <fieldset>
          <legend>
            {label}
          </legend>
          {children}
        </fieldset>
      : <div>
          {children}
        </div>}
  </Form>
