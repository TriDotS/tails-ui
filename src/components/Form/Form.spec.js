import Form from './Form'
import React from 'react'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  render(<Form />, div)
})

it('generate from snapshot', () => {
  const component = renderer.create(<Form />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
