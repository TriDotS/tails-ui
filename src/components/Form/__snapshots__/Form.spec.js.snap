// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate from snapshot 1`] = `
<form
  className="sc-bdVaJa eVqmiX"
  onSubmit={undefined}
>
  <div />
</form>
`;
